src/Module/OStatus/PortableContacts.php

```diff
@@ -0,0 +1,278 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * @see https://web.archive.org/web/20160405005550/http://portablecontacts.net/draft-spec.html
+ */
+
+namespace Friendica\Module\OStatus;
+
+use Friendica\App;
+use Friendica\BaseModule;
+use Friendica\Content\Text\BBCode;
+use Friendica\Core\Cache\Capability\ICanCache;
+use Friendica\Core\Config\Capability\IManageConfigValues;
+use Friendica\Core\L10n;
+use Friendica\Core\Protocol;
+use Friendica\Core\System;
+use Friendica\Database\Database;
+use Friendica\Module\Response;
+use Friendica\Network\HTTPException;
+use Friendica\Util\DateTimeFormat;
+use Friendica\Util\Profiler;
+use Psr\Log\LoggerInterface;
+
+class PortableContacts extends BaseModule
+{
+	/** @var IManageConfigValues */
+	private $config;
+	/** @var Database */
+	private $database;
+	/** @var ICanCache */
+	private $cache;
+
+	public function __construct(ICanCache $cache, Database $database, IManageConfigValues $config, L10n $l10n, App\BaseURL $baseUrl, App\Arguments $args, LoggerInterface $logger, Profiler $profiler, Response $response, array $server, array $parameters = [])
+	{
+		parent::__construct($l10n, $baseUrl, $args, $logger, $profiler, $response, $server, $parameters);
+
+		$this->config   = $config;
+		$this->database = $database;
+		$this->cache    = $cache;
+	}
+
+	protected function rawContent(array $request = [])
+	{
+		if ($this->config->get('system', 'block_public') || $this->config->get('system', 'block_local_dir')) {
+			throw new HTTPException\ForbiddenException();
+		}
+
+		if ($this->args->getArgc() > 1) {
```

Comment:
Is this still necessary because we already set it inside the route?

tbh I do want to get rid of these `$this->args` calls, because they seems like a legacy way to "find" the right data

Reply:
This is the weirdest thing. we generate `/poco/<nickname>/` URLs in the code, but we don't process them at all. Alternatively to this, I could check whether `$this->parameters['nickname']` is set from the route, or I could remove the optional parameter from the route and this condition and it would also throw a "Not Found" exception.

Reply:
I'd say we should remove the parameter, if it isn't valid.
And we should search for these `/poco/<nickname>` calls and replace them.

But maybe @annando knows the reason behind this behaviour ..

Reply:
PoCo had once been used to search for contacts of users. Since this can be done via the Mastodon compatible API, the backend had been mostly deconstructed.

BTW: PoCo has nothing to do with OStatus at all. It is it's own protocol.

Reply:
I know, just like PubSubHubbub, but they are only supported in Friendica because they were part of the OStatus stack. When we don't want to support OStatus anymore, then they will go as well.

Reply:
No, PoCo is not part of OStatus at all. It is used independently from that.

Reply:
It's definitely part of it, the question is whether we're using it independently ourselves.

Reply:
The PoCo-endpoint was used for contact discovery between DFRN systems. It was never used by GNU Social itself which can be called the "master reference".